src/Microsoft.ML.Featurizers/DateTimeTransformer.cs

```diff
@@ -0,0 +1,782 @@
+            => new DateTimeTransformerEstimator(CatalogUtils.GetEnvironment(catalog), inputColumnName, columnPrefix, columnsToDrop, country);
+
+        #region ColumnsProduced static extentions
+
+        internal static Type GetRawColumnType(this DateTimeTransformerEstimator.ColumnsProduced column)
+        {
+            switch (column)
+            {
+                case DateTimeTransformerEstimator.ColumnsProduced.Year:
+                case DateTimeTransformerEstimator.ColumnsProduced.YearIso:
+                    return typeof(int);
+                case DateTimeTransformerEstimator.ColumnsProduced.DayOfYear:
+                case DateTimeTransformerEstimator.ColumnsProduced.WeekOfMonth:
+                    return typeof(ushort);
+                case DateTimeTransformerEstimator.ColumnsProduced.MonthLabel:
+                case DateTimeTransformerEstimator.ColumnsProduced.AmPmLabel:
+                case DateTimeTransformerEstimator.ColumnsProduced.DayOfWeekLabel:
+                case DateTimeTransformerEstimator.ColumnsProduced.HolidayName:
+                    return typeof(ReadOnlyMemory<char>);
+                default:
+                    return typeof(byte);
+            }
+        }
+
+        #endregion
+    }
+
+    /// <summary>
+    /// The DateTimeTransformerEstimator splits up a date into all of its sub parts as individual columns. It generates these fields with a user specified prefix:
+    /// int Year, byte Month, byte Day, byte Hour, byte Minute, byte Second, byte AmPm, byte Hour12, byte DayOfWeek, byte DayOfQuarter,
+    /// ushort DayOfYear, ushort WeekOfMonth, byte QuarterOfYear, byte HalfOfYear, byte WeekIso, int YearIso, string MonthLabel, string AmPmLabel,
+    /// string DayOfWeekLabel, string HolidayName, byte IsPaidTimeOff
+    ///
+    /// You can optionally specify a country and it will pull holiday information about the country as well
+    /// </summary>
+    /// <remarks>
+    /// <format type="text/markdown"><![CDATA[
+    ///
+    /// ###  Estimator Characteristics
+    /// |  |  |
+    /// | -- | -- |
+    /// | Does this estimator need to look at the data to train its parameters? | No |
+    /// | Input column data type | Int64 |
+    /// | Output column data type | Columns and types listed in the summary |
+    ///
+    /// The <xref:Microsoft.ML.Transforms.CategoryImputerEstimator> is a trivial estimator and does not need training.
+    ///
+    ///
+    /// ]]>
+    /// </format>
+    /// </remarks>
+    /// <seealso cref="DateTimeTransformerExtensionClass.DateTimeTransformer(TransformsCatalog, string, string, DateTimeTransformerEstimator.ColumnsProduced[])"/>
+    /// <seealso cref="DateTimeTransformerExtensionClass.DateTimeTransformer(TransformsCatalog, string, string, DateTimeTransformerEstimator.ColumnsProduced[], DateTimeTransformerEstimator.Countries)"/>
+    public sealed class DateTimeTransformerEstimator : IEstimator<DateTimeTransformer>
```

Comment:
The term `TransformerEstimator` is confusing. An object is either a "transformer" or it is an "estimator". There isn't such a thing as a "TransformerEstimator".

Reply:
Changed to just be `DateTimeEstimator`.

---
In reply to: [353904119](https://github.com/dotnet/machinelearning/pull/4521#discussion_r353904119) [](ancestors = 353904119)